libpod/runtime_pod_linux.go

```diff
@@ -109,6 +109,19 @@ func (r *Runtime) NewPod(ctx context.Context, options ...PodCreateOption) (*Pod,
 		if err := pod.save(); err != nil {
 			return nil, err
 		}
+
+		// Once the pod infra container has been created, we start it
+		if err := ctr.Start(ctx); err != nil {
+			// If the infra container does not start, we need to remove the container
+			// and tear the pod down.
+			if err2 := r.RemoveContainer(ctx, ctr, true); err != nil {
```

Comment:
Lowercase removeContainer() so we don't run into a deadlock here

Reply:
Actually - I don't think there's a point to doing this separately if we're force-removing the pod below - it should get rid of the containers in the pod, so we don't need an explicit call to RemoveContainer?

Reply:
Ok, fixed up!